controllers/grid/languages/LanguageGridHandler.inc.php

```diff
@@ -103,8 +103,22 @@ function saveLanguageSetting($args, $request) {
 						$key = array_search($locale, $supportedSubmissionLocales);
 						if ($key !== false) unset($supportedSubmissionLocales[$key]);
 						$supportedSubmissionLocales = array_values($supportedSubmissionLocales);
+						if ($supportedSubmissionLocales == []) {
+							return new JSONMessage(false, __('notification.localeSettingsCannotBeSaved'));
+						}
 						$context = $contextService->edit($context, ['supportedSubmissionLocales' => $supportedSubmissionLocales], $request);
 					}
+
+					if ($settingName == 'supportedSubmissionLocales') {
```

Comment:
I think this check is added too late -- after the context is updated.

Reply:
We've made some test scenarios where tried to disable all submissions checkboxes, everyone received the locale settings cannot be saved notification.

@NateWr we could do a double check here and see if there's any other scenario to test, what do you think?

Reply:
I think the use-case that Alec has identified here is when the supported submission locales are disabled automatically as part of the form locales: https://github.com/pkp/pkp-lib/blob/f704ad0a67bbf12f79cdd67db2509ed25bd73c7d/controllers/grid/languages/LanguageGridHandler.inc.php#L100-L110

So an issue might arise in the following case:

1. en_US enabled for all locales
2. fr_CA enabled for UI + Form but not submission
3. Disable Form locales for en_US

In such a case, the journal may end up with no supported submission locales. Can you test that out?